src/com/cft/contactmerge/tests/CompareContactPartsTest.java

```diff
@@ -31,43 +30,648 @@ void doNamesMatch_Yes_IgnoreSpaces() {
+    }
+
+    @Test
+    void doNamesMatch_No_IgnoreMultiPunctuationInFirstAndLastName_DifferentLastNames() {
+        assertEquals(AnswerType.no, CompareContactParts.doNamesMatch("An-De-Denise", "Sa-la-Democh", "An De Denise", "Sa le Democh"));
+    }
+
+    @Test
+    void doNamesMatch_No_IgnoreMultiPunctuationInFirstAndLastName_CaseInsensitive_DifferentLastNames() {
+        assertEquals(AnswerType.no, CompareContactParts.doNamesMatch("aN-De-denIse", "Sa-LA-Democh", "An dE Denise", "sa le democh"));
+    }
+
+    @Test
+    void doNamesMatch_No_IgnoreMultiPunctuationInFirstAndLastName_CaseInsensitive_IgnoreSpaces_DifferentLastNames() {
+        assertEquals(AnswerType.no, CompareContactParts.doNamesMatch(" aN-De-denIse", " Sa-LA-Democh", "   An dE Denise  ", "sa le democh  "));
     }
 
     @Test
-    void doNamesMatch_Maybe_NamesContainMultiHyphens() {
-        assertEquals(AnswerType.maybe, CompareContactParts.doNamesMatch("Areb", "Sa-la-Democh", "Areb", "Democh"));
+    void doNamesMatch_No_IgnoreMultiPunctuationInFirstAndLastName_DifferentFirstAndLastNames() {
+        assertEquals(AnswerType.no, CompareContactParts.doNamesMatch("An-Day-Denise", "Sa-la-Democh", "An De Denise", "Sa le Democh"));
     }
 
+    @Test
+    void doNamesMatch_No_IgnoreMultiPunctuationInFirstAndLastName_CaseInsensitive_DifferentFirstAndLastNames() {
+        assertEquals(AnswerType.no, CompareContactParts.doNamesMatch("aN-Day-denIse", "Sa-LA-Democh", "An dE Denise", "sa le democh"));
+    }
+
+    @Test
+    void doNamesMatch_No_IgnoreMultiPunctuationInFirstAndLastName_CaseInsensitive_IgnoreSpaces_DifferentFirstAndLastNames() {
+        assertEquals(AnswerType.no, CompareContactParts.doNamesMatch(" aN-Day-denIse", " Sa-LA-Democh", "   An dE Denise  ", "sa le democh  "));
+    }
+
+    //-------------------------------------------------------------------------- Hyphenated First Names -------------------------------------------------------------------------
+    // If the name is less than 4 characters and is contained inside the match name, return maybe.
+    // If the name is 4 or more characters and is contained inside the match name, return yes.
+    @Test
+    void doNamesMatch_YesMaybe_HyphenatedFirstNames() {
+        assertEquals(AnswerType.maybe, CompareContactParts.doNamesMatch("An-Denise", "Doe", "An", "Doe"));
+        assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch("An-Denise", "Adams", "Denise", "Adams"));
+        assertEquals(AnswerType.maybe, CompareContactParts.doNamesMatch("An", "Doe", "An-Denise", "Doe"));
+        assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch("Denise", "Adams", "An-Denise", "Adams"));
+    }
+
+    @Test
+    void doNamesMatch_YesMaybe_HyphenatedFirstNames_CaseInsensitive() {
+        assertEquals(AnswerType.maybe, CompareContactParts.doNamesMatch("An-Denise", "Doe", "aN", "Doe"));
```

Comment:
Some argue you should never have more than one assert in a single test. I don't agree with that. However, I do believe that the test must be structured so that when a test fails, you know why it failed. I don't believe we will know in these cases because we have multiple asserts with the same expected return type. I have handled this in C# by adding a message indicating what test failed but not sure if JUnit supports that.

Reply:
I will research if there is a way to add default messages for failing tests using junit.

Reply:
Could Assert.That() solve the problem? I still haven't had a chance to read the article you sent me.

Reply:
If you insert a string as the 3rd argument in "assertEquals"... this string will be the error message printed when the test fails. Example...
https://i.imgur.com/VINMZsH.png